### DISCUSSION TITLE
Initializing textEditingController using provider

### DISCUSSION THREAD

**Discussion by sabin26** [rrousselGit/riverpod#2342] [Q&A]
Currently, I am defining and disposing text controllers using providers. This removed the need to use TextField.onChanged method since the controller could now be accessed and listened from other providers.

```dart
@riverpod
TextEditingController title(final TitleRef ref) {
  final titleController = TextEditingController();
  ref.onDispose(titleController.dispose);
  return titleController;
}
```

Though it's so much easier, 'riverpod_lint' is not happy with it. It states: 'The 'riverpod_generator' package does not support ChangeNotifier values'. 

I looked into it and could find two alternatives:
1. Using ConsumerStatefulWidget (initialize and dispose controller similar to stateful widgets)
2. Using hooks (useTextEditingController)

Is there any approach for using it inside providers? Or should I just refactor it with one of the alternatives.

**Comment by motucraft** [comment]
Can't it simply be a function?

```dart
TextEditingController createTitleController(Ref ref) {
  final controller = TextEditingController();
  ref.onDispose(controller.dispose);
  return controller;
}
```

**Reply by sabin26** [reply]
No ! It can't. The reason is well explained already on the root of this discussion.
